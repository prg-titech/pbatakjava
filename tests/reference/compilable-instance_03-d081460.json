{
    "basename": "compilable-instance_03-d081460",
    "cmd": "{infile}",
    "infile": "tests/instance_03/build.sh",
    "infile_hash": "abf159e9e18f849984fa106090fd5f92a820c2e96110f69d0850172b",
    "stdout": "compilable-instance_03-d081460.stdout",
    "stdout_hash": "1b63061c8375b2e0bcfda6c7071aa94f3ecbb22b9802c641e483c6c5",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}