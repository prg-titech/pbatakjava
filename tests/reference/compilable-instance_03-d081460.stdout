
public interface Point {
}

public interface Point_Factory {
  public Point make(int x, int y);
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public void m() {
    return;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }
}

public class Point_v2 implements Point {
  public int x;

  public int y;

  public Point_v2(int x, int y) {
    this.x = x;
    this.y = y;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v2(x, y);
  }
}

public interface Line {
}

public interface Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2);
}

public class Line_v1 implements Line {
  public Point_Factory V;

  public Point_Factory W;

  public Point p1;

  public Point p2;

  public Line_v1(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    this.V = V;
    this.W = W;
    this.p1 = p1;
    this.p2 = p2;
  }
}

public class Line_v1_Factory implements Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    return new Line_v1(V, W, p1, p2);
  }
}

public interface Test {
}

public interface Test_Factory {
}

public class Test_v1 implements Test {
  public static void main(String[] args) {
    Point_v1 p1 = new Point_v1(1, 1);
    p1.m();
    Line_v1 l1 = new Line_v1(new Point_v1_Factory(), new Point_v2_Factory(), p1, new Point_v2(2, 2));
  }
}

public class Test_v1_Factory implements Test_Factory {
}
